### PR TITLE
Allow hiding top panel via blueprint

### PR DESCRIPTION
**Pull request by jprochazk**
### What

- `rrb.TopPanel` is now exposed, works similarly to `rrb.TimePanel`
- `TimePanel` hidden state now fully hides the time panel

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6409?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6409?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6409)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Review by Wumpf — CHANGES_REQUESTED** [review]
doesn't actually work ;)

Bunch of things we need to do:
* process the incoming `TopPanel`
* extend docs:
    * what does collapsed vs hidden mean on each of the panels
    * what does collapse_all thingy bool actually do now
    * ... stuff? 


How I tested:
`pixi run python ./docs/snippets/all/views/spatial2d.py`
`pixi run ipython`
then
```py
import rerun.blueprint as rrb
rrb.Blueprint(rrb.TopPanel(state="collapsed")).connect("rerun_example_spatial_2d")
```

File: crates/re_viewer/src/app.rs [hunk 1]
```diff
@@ -884,7 +884,7 @@ impl App {
 
                 crate::ui::mobile_warning_ui(&self.re_ui, ui);
 
-                if app_blueprint.top_panel_state != PanelState::Hidden {
+                if app_blueprint.top_panel_state == PanelState::Expanded {
```
Comment (jprochazk) [inline crates/re_viewer/src/app.rs:887]: this is to align with behavior in https://github.com/rerun-io/rerun/pull/6419

File: rerun_py/rerun_sdk/rerun/blueprint/api.py [hunk 2]
```diff
@@ -335,6 +337,8 @@ def __init__(self, *, expanded: bool | None = None, state: PanelStateLike | None
         state:
             Whether the panel is expanded, collapsed, or hidden.
 
+            Collapsed and hidden both fully hide the top panel.
```
Comment (Wumpf) [inline rerun_py/rerun_sdk/rerun/blueprint/api.py:340]:
```suggestion
            Collapsed and hidden both fully hide the top panel.
```
```suggestion
            Collapsed and hidden both fully hide the blueprint panel.
```

File: rerun_py/rerun_sdk/rerun/blueprint/api.py [hunk 3]
```diff
@@ -353,6 +357,8 @@ def __init__(self, *, expanded: bool | None = None, state: PanelStateLike | None
...
```
Comment (Wumpf) [inline rerun_py/rerun_sdk/rerun/blueprint/api.py:360]:
```suggestion
            Collapsed and hidden both fully hide the top panel.
```
```suggestion
            Collapsed and hidden both fully hide the selection panel.
```

**Review by Wumpf — CHANGES_REQUESTED** [review]
On native mac the hidden top panel doesn't work out, I think we have to do a per-platform thing here

when selected:
image

when unselected:
image

I figure on Windows and Linux this is fine because we don't mange the close buttons there. Also, for web this is obviously a good feature to have (maybe worth noting as well :)).
I'd propose to disable this on Mac (needs to be documented) and makes sure this looks alright on Windows & Linux.

**Comment by jprochazk** [comment]
> I'd propose to disable this on Mac (needs to be documented) and makes sure this looks alright on Windows & Linux.

Instead of disabling it, I made the top panel toggle hide its content instead of the full thing on mac. Currently making sure it looks alright everywhere

**Comment by Wumpf** [comment]
good idea! Confirmed on my side this looks alright on Mac

**Review by Wumpf — APPROVED** [review]
At first I thought "fully hides the panel" isn't quite right on Mac now, but arguably it still is as we only have the bare minimum title bar that the OS needs essentially (we could shrink it a bit down with only having the mac buttons in there but whatever, can do that later if we _really_ want to optimize this), so let's keep things simple!

Great stuff and thanks for all the followups!

**Pull request merged**